File: app/src/main/java/camerakit/android/app/MainActivity.java
package camerakit.android.app;

import android.app.Activity;
import android.os.Bundle;

import camerakit.android.CameraKitView;

public class MainActivity extends Activity {

    private CameraKitView camera;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        camera = findViewById(R.id.camera);
    }

    @Override
    protected void onStart() {
        super.onStart();
        camera.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
        camera.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        camera.onPause();
    }

    @Override
    protected void onStop() {
        super.onStop();
        camera.onStop();
    }

}

File: demo/src/main/java/camerakit/android/demo/MainActivity.kt
package camerakit.android.demo

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem

class MainActivity : AppCompatActivity() {

    private var pagerAdapter: PagerAdapter? = null
    private val cameraFragment: CameraFragment = CameraFragment()
    private val galleryFragment: GalleryFragment = GalleryFragment()

    private val fragments = arrayOf(cameraFragment, galleryFragment)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        pagerAdapter = PagerAdapter(supportFragmentManager)

        val viewPager = findViewById<ViewPager>(R.id.viewPager)
        val tabLayout = findViewById<TabLayout>(R.id.tabLayout)

        viewPager.adapter = pagerAdapter
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))

        tabLayout.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(viewPager))
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.action_info) {
            AlertDialog.Builder(this)
                    .setTitle(getString(R.string.info_title, BuildConfig.VERSION_NAME))
                    .setMessage(R.string.info_message)
                    .setPositiveButton(R.string.info_button, null)
                    .setCancelable(true)
                    .show()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    inner class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment {
            return fragments[position]
        }

        override fun getCount(): Int {
            return fragments.size
        }

    }

}
